### ISSUE TITLE
[data grid] Cannot provide autocomplete options to a datagrid multi-select filter operation input component

### ISSUE THREAD

**Issue by dhaub-exelixis** [mui/mui-x#9515]
### What's the problem? 🤔

I have a use case where I'd like our users to be able to filter some columns that have a relatively small number of known values without needing to type the desired value. Looking at the definition for the component that is used to collect input for filter operations like ~isOneOf~ `isAnyOf`, the ability to provide a set of options is completely disabled. It's interesting to note that it's still possible to turn off freeSolo

It looks like that prop is explicitly blocked from being passed to the component here: https://github.com/mui/mui-x/blob/50f9125d95332883f68c88de1fe162893f25b021/packages/grid/x-data-grid/src/components/panel/filterPanel/GridFilterInputMultipleValue.tsx#L11C61-L11C61

Is there a reason for removing this ability that I'm missing, or is it possible to have that ability restored? This would be incredibly helpful for my project. I know it is reasonable to define one's own input component, but with the ability to provide a list of options, the solution to this problem would be much cleaner.

### What are the requirements? ❓

- `options` can be passed successfully in the `InputComponentProps` property of a multi-select filter operation

### What are our options? 💡

Defining a whole new InputComponent specifically to allow a list of options to be passed to the Autocomplete component

### Proposed solution 🟢

I don't have the full context around this part of DataGrid, but it seems simply removing `options` from the omit line would be enough to add this feature  https://github.com/mui/mui-x/blob/50f9125d95332883f68c88de1fe162893f25b021/packages/grid/x-data-grid/src/components/panel/filterPanel/GridFilterInputMultipleValue.tsx#L11C61-L11C61

**Comment by dhaub-exelixis** [comment]
Oh that's my bad, I wasn't the one on my team who originally set up the column definitions and to be honest when I was working on filtering yesterday I got so in my own head I didn't even think to look there! I just saw the `InputComponentProps` field in the filtering docs and figured since it was an Autocomplete component, passing `options` would work. 

I just got the options working correctly using the `GridFilterInputMultipleSingleSelect` component. Thank you for pointing me in the right direction!

```
      filterOperators: (() => {
        const operator = getGridStringOperators().filter((o) => o.value === "isAnyOf")[0];
        operator.InputComponent = GridFilterInputMultipleSingleSelect;
        return [operator];
      })(),
```

I am still a little confused about `GridFilterInputMultipleValue.` That component specifically doesn't seem to pick up on the `valueOptions` prop either (I am using v5.17.12 so if that's been updated, I aplolgize). Wouldn't it make sense to be able to override the `options` prop if needed?
